src/screenComponents/miniGame.h

```diff
@@ -0,0 +1,40 @@
+/** An abstract class to present an interface on which to implement (hacking)
+ *  minigames. Concrete implementations will need to implement at least their
+ *  own constructor and reset() functions, and probably the disable() function.
+ *  Calling gameComplete() registers a successful minigame with the owner
+ */
+
+#ifndef MINIGAME_H
+#define MINIGAME_H
+
+#include <vector>
+#include "hackingDialog.h"
+
+class GuiButton;
+class GuiToggleButton;
+class GuiProgressbar;
+class GuiLabel;
+class GuiPanel;
+
+class MiniGame
```

Comment:
Would be good to add a base class of sf::NonCopyable:
https://www.sfml-dev.org/documentation/2.5.1/classsf_1_1NonCopyable.php

Reply:
Done